### PR TITLE
ethereum: ensure forge tests work with new forge

### PR DESCRIPTION
**Pull request by SEJeff**
Otherwise, you get errors like:

    Compiler run failed:
    Error (8961): Function cannot be declared as view because this expression (potentially) modifies the state.
    --> forge-test/TokenImplementation.t.sol:84:26:
    |
    84 |         output.allower = vm.addr(sk);
    |                          ^^^^^^^^^^^

    Error (8961): Function cannot be declared as view because this expression (potentially) modifies the state.
    --> forge-test/TokenImplementation.t.sol:101:42:
        |
    101 |         (output.v, output.r, output.s) = vm.sign(sk, message);
        |                                          ^^^^^^^^^^^^^^^^^^^^

    Error (8961): Function cannot be declared as view because this expression (potentially) modifies the state.
    --> forge-test/relayer/WormholeSimulator.sol:335:45:
        |
    335 |         (sigs[0].v, sigs[0].r, sigs[0].s) = vm.sign(devnetGuardianPK, vm_.hash);
        |                                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    Error (8961): Function cannot be declared as pure because this expression (potentially) modifies the state.
    --> forge-test/relayer/WormholeRelayer.t.sol:166:9:
        |
    166 |         vm.assume(gasParams.evmGasOverhead > 0);
        |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    Error (8961): Function cannot be declared as pure because this expression (potentially) modifies the state.
    --> forge-test/relayer/WormholeRelayer.t.sol:167:9:
        |
    167 |         vm.assume(gasParams.targetGasLimit > 0);
        |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    Error (8961): Function cannot be declared as pure because this expression (potentially) modifies the state.
    --> forge-test/relayer/WormholeRelayer.t.sol:168:9:
        |
    168 |         vm.assume(feeParams.targetNativePrice > 0);
        |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    Error (8961): Function cannot be declared as pure because this expression (potentially) modifies the state.
    --> forge-test/relayer/WormholeRelayer.t.sol:169:9:
        |
    169 |         vm.assume(gasParams.targetGasPrice > 0);
        |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

    Error (8961): Function cannot be declared as pure because this expression (potentially) modifies the state.
    --> forge-test/relayer/WormholeRelayer.t.sol:170:9:
        |
    170 |         vm.assume(gasParams.sourceGasPrice > 0);
        |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

### CODE REVIEW

**Comment by kcsongor** [comment]
relevant: #3086

tldr; is it possible that you have an *outdated* version of forge-std?

**Comment by SEJeff** [comment]
@kcsongor I'm trying to get a PR that used the old version to the latest main so it can be merged, so yes that appears to be it. Thanks
